### DISCUSSION TITLE
Issue with playback of large size files

### DISCUSSION THREAD

**Discussion by kapilggg10** [asciinema/discussions#230] [General]
Let's assume I have a .cast file of 1 or 2 GB, Asciinema-player fetch the whole file at once which takes a lot of time to fetch these large files, and meanwhile Asciinema-player is not showing any loading or fetching message.

1. We should have buffer strategy to play video instead of loading the file at once.
2. Some caching mechanism can be implemented or re-creation of player ca be avoided to avoid re-fetching of file if we change speed option in the props with the same url.

**Comment by FriendlyTroll** [comment]
Same issue here. I use asciinema to recored my daily work in terminal. This sometimes results in large files. I am trying to play a cast of 2GB and the player just hangs. Playing the cast from terminal works fine but there I can't scroll through the video.

Is there a way maybe to separate the cast into smaller files? I would just like to replay a part of the cast towards the end of the recording.

**Comment by ku1ik** [comment]
There's no easy way of solving that, it would need major refactoring of the player's `recording` driver, so I've turned the original issue into a discussion here.

**Comment by FriendlyTroll** [comment]
Hi,

a workaround is to split the file to multiple smaller files example:

`split -b 50M --additional-suffix=.cast 2023-05-31_15-14-30.cast`

This will create multiple 50M files which you can then play, but make sure to format the each file correctly. At the beginning on first line there should be a header

```
`{"version": 2, "width": 211, "height": 55, "timestamp": 1685538870, "idle_time_limit": 2.0, "env": {"SHELL": "/bin/bash", "TERM": "xterm-256color"}}`
```

Also make sure the second and last line is properly formatted, probably the easiest is to just delete those lines if they are missing parenthesis or similar.
